### PR TITLE
Added support for OLI events from xblocks.

### PR DESCRIPTION
**Pull request by dcadams**
@stvstnfrd @jbassen

### CODE REVIEW

**Comment by dcadams** [comment]
@stvstnfrd @jbassen This is ready for review now.

This is part of the urgent work need for the OLI go-live. I attempted to write a unit test for the new method but it proved to be a bit tricky and I haven't got it running yet. I'll add a unit test in a later PR.

File: common/djangoapps/track/backends/oli.py [hunk 1]
```diff
@@ -78,10 +82,18 @@ def send(self, event):
...
+        # Look where it should be for a capa prob.
         problem_id = event_data.get('problem_id')
+        if problem_id:
+            problem_text = self.get_problem_text(problem_id, course_id)
+
         if not problem_id:
-            LOG.info('problem_id attribute missing from event for OLI service')
-            return None
+            # Look where it should be for an xblock.
+            problem_id = context.get('module').get('usage_key')
```
Comment (stvstnfrd) [inline common/djangoapps/track/backends/oli.py]: So it looks like if we need to grab the `problem_id` from here,
we don't refetch the `problem_text`; should we?

File: common/djangoapps/track/backends/oli.py [hunk 2]
```diff
@@ -136,3 +149,22 @@ def send(self, event):
...
+    def get_problem_text(self, block, course_id):
+        """
+        Helper method to get the problem text
+        """
+
+        course_key = CourseKey.from_string(course_id)
+        usage_key = UsageKey.from_string(block).map_into_course(course_key)
+
+        try:
+            descriptor = modulestore().get_item(usage_key, depth=5)
```
Comment (stvstnfrd) [inline common/djangoapps/track/backends/oli.py]: Why `depth=5`?

**Comment by stvstnfrd** [comment]
2 questions, otherwise looks good.

**Comment by stvstnfrd** [comment]
+1

**Pull request merged**